flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/strategy/LazyFromSourcesSchedulingStrategyTest.java

```diff
@@ -377,4 +377,16 @@ private LazyFromSourcesSchedulingStrategy startScheduling(TestingSchedulingTopol
 		schedulingStrategy.startScheduling();
 		return schedulingStrategy;
 	}
+
+	private static List<ExecutionVertexID> mapExecutionVerticesToIDs(
+			final List<TestingSchedulingExecutionVertex> vertices) {
+
+		return vertices.stream().map(TestingSchedulingExecutionVertex::getId).collect(Collectors.toList());
+	}
+
+	private static List<ExecutionVertexID> mapDeploymentOptionsToIDs(
```

Comment:
If you want to stick with `assertEquals`, I am fine with that. However, I would add a method to `TestingSchedulerOperations` that directly returns the execution vertex ids to avoid having `mapDeploymentOptionsToIDs()` calls everywhere.

Reply:
You are right we should try to avoid spreading the `mapDeploymentOptionsToIDs ` calls.
A `checkLatestScheduledVertices()` is added to make the check simpler.